### DISCUSSION TITLE
Proxy requests in `createResource`

### DISCUSSION THREAD

**Discussion by daalfox** [solidjs/solid-start#1480] [Q&A] [answered]
How should I use fetch api with a proxy? I.e I want to do something like `fetch('/api/etc')` inside the callback argument of `createResource`
I searched a lot but there are a lot of options about proxies in config that I got confused.

**Comment by atk** [comment]
This is impossible, since fetch can only do http(s) requests, whereas a proxy requires a different protocol to be used. Proxies can be configured in the browser settings. 

However, what you most likely want is a local tunnel server that allows you to bypass cors restrictions, right?

**Reply by daalfox** [reply]
yes exactly

**Reply by atk** [reply]
In Solid-Start, you can use an [API route](https://docs.solidjs.com/solid-start/building-your-application/api-routes) to create such a tunnel yourself.

**Reply by daalfox** [reply]
but I don't want to expose the route

**Reply by atk** [reply, marked as answer]
Either it is exposed or even your own site cannot use it. Make sure it is as limited as possible, CORS headers set to be only used from your site, etc.
